src/infrastructure/app/AppWrapper.ts

```diff
@@ -40,8 +40,12 @@ export default class AppWrapper {
 
   constructor(controllers?: BaseController[]) {
     this.setup();
-    this.apiDocGenerator = new ApiDocGenerator(AppSettings.Environment, config.Params.ApiDocsInfo);
     this.app = AppServer();
+    this.apiDocGenerator = new ApiDocGenerator(AppSettings.Environment, config.Params.ApiDocsInfo);
+    this.apiDocGenerator.setServer(

```

Comment:
I'm not sure about this line here.
It's server context, maybe it's better where it was and you could do something like this: 

```ts
// ApiDocGenerator
  saveApiDoc(): void {
    const wasDocGenerated = !(this.env !== DEV || !Object.keys(this.apiDoc.paths).length);
    if (wasDocGenerated) {
      const filePath = resolve(join(__dirname, "../../../../openapi.json"));
      writeFileSync(filePath, JSON.stringify(this.apiDoc, null, 2), "utf8");
    }

   // Is important to clean the garbage
    SchemasStore.dispose();
  }

  setServer(url: string, description: "Local server"): this {
    this.apiDoc.servers.push({
      url,
      description,
    });

    this.setSchemas(SchemasStore.get());
  }

// In HttpServer
    this.server.on("listening", () => {
      this.#appWrapper.apiDocGenerator
        .setServer(`http://${AppSettings.getServerUrl()}`, "Local server")
        .saveApiDoc();
      console.log(`Server ${AppSettings.ServiceName} running on ${AppSettings.getServerUrl()}`);
    });

// In AppSettings
  static getServerUrl(): string {
    return `${this.ServerHost}:${this.ServerPort}${this.ServerRoot}`;
  }
```

Reply:
run the lint here and made the commit


I ended up getting excited and implemented the generation of OpenAPI with bearer authentication and raised the same PR, I hadn't seen that you had already checked, sorry

I had made this change to use swagger-ui-express, I even implemented it and was going to suggest it in a next Pull Request

I'll think about the adjustments here.

If you want, I can send PR's for each feature suggestion so you don't end up with a very big PR

Congratulations for your work!

Reply:
Thank you for your congratulations @retamiro and feel free to send your contributions, they are welcome and I will surely receive them with great pleasure and review them with great care. 

About swagger-ui-express of course, we can start to using it and you are right, little by little.  😊👽